tests/dags/common/sensors/test_single_run_external_dags_sensor.py

```diff
@@ -0,0 +1,219 @@
+import logging
+import unittest
+
+import pytest
+from airflow.exceptions import AirflowException
+from airflow.models import DagBag, DagRun, Pool, TaskInstance
+from airflow.models.dag import DAG
+from airflow.utils.db import add_default_pool_if_not_exists
+from airflow.utils.session import create_session
+from airflow.utils.state import State
+from airflow.utils.timezone import datetime
+from airflow.utils.types import DagRunType
+from common.sensors.single_run_external_dags_sensor import SingleRunExternalDAGsSensor
+
+
+DEFAULT_DATE = datetime(2022, 1, 1)
+TEST_TASK_ID = "wait_task"
+TEST_POOL = "single_run_external_dags_sensor_test_pool"
+DEV_NULL = "/dev/null"
+
+
+@pytest.fixture(autouse=True)
+def clean_db():
+    with create_session() as session:
+        session.query(DagRun).delete()
+        session.query(TaskInstance).delete()
+        session.query(Pool).delete()
+        add_default_pool_if_not_exists(session)
+
+
+def run_sensor(sensor):
+    sensor.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
+
+
+def create_task(dag, task_id, external_dag_ids):
+    return SingleRunExternalDAGsSensor(
+        task_id=task_id,
+        external_dag_ids=[],
+        check_existence=True,
+        dag=dag,
+        pool=TEST_POOL,
+        poke_interval=5,
+        mode="reschedule",
+    )
+
+
+def create_dag(dag_id, task_id=TEST_TASK_ID):
+    with DAG(
+        dag_id,
+        default_args={
+            "owner": "airflow",
+            "start_date": DEFAULT_DATE,
+        },
+    ) as dag:
+        # Create a sensor task inside the DAG
+        create_task(dag, task_id, [])
+
+    return dag
+
+
+def create_dagrun(dag, dag_state):
+    return dag.create_dagrun(
+        run_id=f"{dag.dag_id}_test",
+        start_date=DEFAULT_DATE,
+        execution_date=DEFAULT_DATE,
+        data_interval=(DEFAULT_DATE, DEFAULT_DATE),
+        state=dag_state,
+        run_type=DagRunType.MANUAL,
+    )
+
+
+class TestExternalDAGsSensor(unittest.TestCase):
```

Comment:
I wasn't aware you could mix unittest.TestCase classes with pytest stuff, TIL!

Reply:
Pytest is super robust this way, thankfully! I think it would be feasible to turn this into function-based tests (this was based off of Airflow's own tests which are class structured) but that's totally up to @stacimc :slightly_smiling_face:

Reply:
I guess unittest.TestCase has some nice stuff Staci is using like assertLogs and such. `setUp` I guess would just be an auto-used fixture?

Reply:
Yea that's exactly what we could do! Although `assertLogs` would be just a hair harder to change, though the [`caplog` fixture](https://docs.pytest.org/en/6.2.x/logging.html#caplog-fixture) makes that easier.

Reply:
I was surprised too! I was using Airflow's own tests as a reference to test the Sensor 😰 and found that we were mixing unittest.TestCase with pytest in some ingestion server tests 🤯